coding-concepts.js

```diff
@@ -8,36 +8,44 @@
 // --------------------1) What will this log?
 
 const cohort = "Golf 2022"
-// console.log(cohort.split(""))
+console.log(cohort.split(""))
+
+// a) Your answer:["G","o","l","f"," ", "2","0","2","2"]
+// b) Verify and explain: [
+//   'G', 'o', 'l',
+//   'f', ' ', '2',
+//   '0', '2', '2'
+// ]
+// the split method will split a string into an array based on what you put in the argument. In this clase it was an empty string so it split after each character including the space which is technically a character. The ascii value of space is 32 or 0010 00002 in binary so that is what the machine is reading so it shows up. The argument actually tells the string where it should split so for example if I put:
+console.log(cohort.split(" "))
+// the outcome would be ["Golf", "2022"]. In this way we can loop through individual words in a long string to check for something like word count of a paragraph.
 
-// a) Your answer:
-// b) Verify and explain:
 
 // --------------------2) What will this log?
 
 const greeter = (name) => {
   `Hello, ${name}!`
 }
-// console.log(greeter("LEARN Student"))
+console.log(greeter("LEARN Student"))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: undefined
+// b) Verify and explain: There is no return so we got an undefined. If a function doesn't have a return statement it will just return undefined by default.
 
 // --------------------3) What will this log?
 
 const multipliedByTwo = [4, 5, 6, 7, 8].map((number) => number * 2)
-// console.log(multipliedByTwo)
+console.log(multipliedByTwo)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: there is no return statement here in the map function so I am tempted to say undefined. I am unsure though because I think there might be an implicit return since it is only one line. If that is the case then it will be [8,10,12,14,16]
+// b) Verify and explain: So there is an implicit return in the .map function. I think it can only be used when the logic can fit on a single line. If the logic requires a new line curly bracers and a return should be used. Though I did find some examples that appeared to have multiple lines and no return so I am unsure. Hard to find documentation on this. I am going to go with if it is easily readable like the above and fits on one line then no return is necessary otherwise I should use curly bracers and a return. The .map function iterates through each element of the array and performs some action on that element. The .map returns an array of equal length as the array passed into it. For now...
```

Comment:
There has been some recently updated code that allows for you to have multiple lines with no curlies and return, but it is an all-or-nothing situation.  The preferred method still is using curlies with a return.